source/community/reactnative/src/models/container-model.js

```diff
@@ -0,0 +1,155 @@
+import { BaseModel } from './base-model'
+import { ModelFactory } from './model-factory';
+import { ElementType } from '../utils/enums'
+import {ImageModel} from './element-model'
+
+class BaseContainerModel extends BaseModel {
+    constructor(parent, payload) {
+        super(parent, payload);
+        if (payload.backgroundImage) {
+            this.backgroundImage = payload.backgroundImage;;
+        }
+        this.verticalContentAlignment = payload.verticalContentAlignment;
+        this.style = payload.style;
+        this.bleed = payload.bleed;
+    }
+}
+
+export class AdaptiveCardModel extends BaseContainerModel {
+    constructor(parent, payload) {
+        super(parent, payload);
+        this.type = ElementType.AdaptiveCard;
+        this.fallbackText = payload.fallbackText;
+        this.version = payload.version;
+        this.speak = payload.speak;
+        this.children = [];
+        this.actions = [];
+        this.children.push(...ModelFactory.createGroup(this, payload.body));
+        this.actions.push(...ModelFactory.createGroup(this, payload.actions));
+        this.show = true;
+    }
+}
+
+export class ContainerModel extends BaseContainerModel {
+    constructor(parent, payload) {
+        super(parent, payload);
+        this.type = ElementType.Container;
+        this.children = [];
+        this.children.push(...ModelFactory.createGroup(this, payload.items));
+        this.height = payload.height;
+    }
+
+    get items(){
+        return this.children;
+    }
+}
+
+export class ColumnSetModel extends BaseContainerModel {
+    constructor(parent, payload) {
+        super(parent, payload);
+        this.type = ElementType.ColumnSet;
+        this.children = [];
+        if (payload.columns) {
+            payload.columns.forEach((item) => {
```

Comment:
Can't we user ModelFactory.createGroup here too?

Reply:
ModelFactory.createGroup makes sense only when the groups contain a varied set of elements like in case of a container. For ColumnSet, FactSet and ImageSet, the items is going to be of a single type.This would avoid the unnecessary condition check and make the expected type clear.

Reply:
Got it.. Makes sense now.